src/main/java/io/druid/data/input/impl/DimensionSchema.java

```diff
@@ -0,0 +1,138 @@
+/*
+* Licensed to Metamarkets Group Inc. (Metamarkets) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. Metamarkets licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.data.input.impl;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.common.base.Preconditions;
+
+import java.util.List;
+
+public class DimensionSchema
+{
+  // This should really use ValueType from main Druid.
+  // TODO: get rid of this when druid-api is merged back into the main repo
+  public enum ValueType
+  {
+    FLOAT,
+    LONG,
+    STRING;
+
+    @JsonValue
+    @Override
+    public String toString()
+    {
+      return this.name().toUpperCase();
+    }
+
+    @JsonCreator
+    public static ValueType fromString(String name)
+    {
+      return valueOf(name.toUpperCase());
+    }
+  }
+
+  private final String name;
+  private final ValueType type;
+  private final boolean isSpatial;
+  private final List<String> subdimensions;
+
+  @JsonCreator
+  public DimensionSchema(
+      @JsonProperty("name") String name,
+      @JsonProperty("type") ValueType type,
+      @JsonProperty("isSpatial") Boolean isSpatial,
```

Comment:
`isSpatial` and `type` are kind of bummers, is it possible at this point to make this polymorphic or is that something we wouldn't be able to do until later?